pilot/adapter/config/file/monitor_test.go

```diff
@@ -78,40 +76,37 @@ func (cm *controllerManager) setup(events chan event) {
 	}
 
 	// Run the controller.
-	go cm.controller.Run(cm.controllerStop)
-}
-
-func (cm *controllerManager) teardown() {
-	close(cm.controllerStop)
+	go cm.controller.Run(stop)
 }
 
 type env struct {
+	stop chan struct{}
 	controllerManager controllerManager
-	monitorCtx        context.Context
-	monitorCancel     context.CancelFunc
 	events            chan event
 	fsroot            string
 	monitor           *file.Monitor
 }
 
 func (e *env) setup(tempDir string) {
 	e.events = make(chan event)
+	e.stop = make(chan struct{})
 
 	// Create and setup the controller.
 	e.controllerManager = controllerManager{}
-	e.controllerManager.setup(e.events)
+	e.controllerManager.setup(e.events, e.stop)
 
-	e.monitorCtx, e.monitorCancel = context.WithCancel(context.Background())
 	e.fsroot = tempDir
 
 	// Make all of the updates go through the controller so that the events are triggered.
 	e.monitor = file.NewMonitor(e.controllerManager.controller, e.fsroot, nil)
-	e.monitor.Start(e.monitorCtx)
+	go e.monitor.Run(e.stop)
+
+	// Sleep for a bit to wait for the server to start.
+	time.Sleep(time.Second)
```

Comment:
is there a better way to do this?

Reply:
That's my question as well ... what is the standard way of solving this in go/istio?